Source/ARTAuthOptions.m

```diff
@@ -135,4 +135,52 @@ - (BOOL)isMethodGET {
     return [_authMethod isEqualToString:@"GET"];
 }
 
+- (ARTAuthOptions *)replaceWith:(ARTAuthOptions *)options {
```

Comment:
To clarify, I mean that instead of this:

```objective-c
- (void)createTokenRequest:(ARTTokenParams *)tokenParams options:(ARTAuthOptions *)options callback:(void (^)(ARTTokenRequest *, NSError *))callback {
	ARTAuthOptions *mergedOptions = [self replaceOptions:options];
	ARTTokenParams *mergedTokenParams = [self replaceParams:tokenParams];
	// ...
```

We can do this:

```objective-c
- (void)createTokenRequest:(ARTTokenParams *)tokenParams options:(ARTAuthOptions *)options callback:(void (^)(ARTTokenRequest *, NSError *))callback {
	options = options ? options : self.options;
	tokenParams = tokenParams ? tokenParams : self.tokenParams;
	// ...
```

And basically get rid of everything else you added (plus the `merge...` methods laying around). Am I missing something?